### PR TITLE
Add Claude Code GitHub Workflow

### PR DESCRIPTION
**Pull request by sindresorhus**
## 🤖 Installing Claude Code GitHub App

This PR adds a GitHub Actions workflow that enables Claude Code integration in our repository.

### What is Claude Code?

[Claude Code](https://claude.com/claude-code) is an AI coding agent that can help with:
- Bug fixes and improvements  
- Documentation updates
- Implementing new features
- Code reviews and suggestions
- Writing tests
- And more!

### How it works

Once this PR is merged, we'll be able to interact with Claude by mentioning @claude in a pull request or issue comment.
Once the workflow is triggered, Claude will analyze the comment and surrounding context, and execute on the request in a GitHub action.

### Important Notes

- **This workflow won't take effect until this PR is merged**
- **@claude mentions won't work until after the merge is complete**
- The workflow runs automatically whenever Claude is mentioned in PR or issue comments
- Claude gets access to the entire PR or issue context including files, diffs, and previous comments

### Security

- Our Anthropic API key is securely stored as a GitHub Actions secret
- Only users with write access to the repository can trigger the workflow
- All Claude runs are stored in the GitHub Actions run history
- Claude's default tools are limited to reading/writing files and interacting with our repo by creating comments, branches, and commits.
- We can add more allowed tools by adding them to the workflow file like:

```
allowed_tools: Bash(npm install),Bash(npm run build),Bash(npm run lint),Bash(npm run test)
```

There's more information in the [Claude Code action repo](https://github.com/anthropics/claude-code-action).

After merging this PR, let's try mentioning @claude in a comment on any PR to get started!

### CODE REVIEW

**Comment by sindresorhus** [comment]
@som-sm Are you ok with this? It could help unburden us with the initial review?

We probably need a CLAUDE.md with some specific instructions.

**Comment by som-sm** [comment]
> @som-sm Are you ok with this? It could help unburden us with the initial review?

Yeah, I’m cool with this! It should help ease the review process, especially for the repetitive parts like docs.

> We probably need a CLAUDE.md with some specific instructions.

Yeah, things like how we do docs, tests etc.

**Comment by sindresorhus** [comment]
Added CLAUDE.md

**Comment by som-sm** [comment]
Should we add something for JSDoc links as well, for issues like [this](https://github.com/sindresorhus/type-fest/pull/1254#discussion_r2404511789).

File: CLAUDE.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+- Count via tuple `['length']`
+- Union distribution is tricky—test it
+
+## Philosophy
+
+- Correctness > cleverness
+- Real problems only
+- Docs teach how, not what
+- Edge cases mandatory
+- One concept per PR
```
Comment (som-sm) [inline CLAUDE.md:108]: Will this be able to catch issues like [this](https://github.com/sindresorhus/type-fest/pull/1254#discussion_r2404555595)? It should, I guess.

Comment (sindresorhus) [reply CLAUDE.md:108]: Made it explicit: https://github.com/sindresorhus/type-fest/pull/1255/files#diff-6ebdb617a8104a7756d0cf36578ab01103dc9f07e4dc6feb751296b9c402faf7R110

**Review by som-sm — APPROVED** [review]
This looks fine!

We'll get to know better once we see it in action, and if required, we can iteratively add more things to `CLAUDE.md`.
